### ISSUE TITLE
Node.duplicate() creates extra nodes under certain conditions

### ISSUE THREAD

**Issue by radlotus** [godotengine/godot#76593]
### Godot version

3.6 beta 1 and 4.0.3 RC1

### Issue description

`Node.duplicate()` doubles the children nodes under certain conditions.
It doesn't occur otherwise.

### Steps to reproduce

1. Take a node that has some children.
2. Pack this node and its children into a PackedScene
3. Save the PackedScene
4. Load the PackedScene and instantiate it
5. Set the `owner` property of the instance's children to `null`
6. Duplicate the node

### Minimal reproduction project

[node_duplicate_creates_extra_nodes_godot3.zip](https://github.com/godotengine/godot/files/11360110/node_duplicate_creates_extra_nodes_godot3.zip)
[node_duplicate_creates_extra_nodes_godot4.zip](https://github.com/godotengine/godot/files/11360111/node_duplicate_creates_extra_nodes_godot4.zip)

**Comment by radlotus** [comment]
Some notes:
- the doubling only occurs on children that had their `owner` property changed
- this occurs only if  `DUPLICATE_USE_INSTANCING` is passed to `Node.duplicate` which is the default
- this occurs only on children of a PackedScene instance loaded using `load()` or `ResourceLoader`. 
- it's not necessary for it to be loaded from a filesystem. it can be loaded from a `resource_path` that's in the cache

**Comment by KoBeWi** [comment]
That's a weird description. What `DUPLICATE_USE_INSTANCING ` does is it calls `instantiate()` on the original PackedScene if it's available. This causes the scene to be re-instantiated.

When you duplicate a scene instance, its children are not duplicated. Instead, the scene is re-instantiated with new children. Whether a node belongs to scene depends on its `owner` property, so when you remove owner, the nodes are no longer part of the scene and get duplicated normally. Thus your duplicate includes the unowned nodes and the nodes from re-instantiation.

tl;dr this is not a bug.
But maybe the documentation can be improved somewhere.

**Comment by Mickeon** [comment]
This would be documented by https://github.com/godotengine/godot/pull/68560.
"_Duplicate using PackedScene.instantiate. If the node comes from a scene saved on disk, re-uses PackedScene.instantiate as the base for the duplicated node and its children._". [(direct link)](https://github.com/godotengine/godot/pull/68560/files#diff-bfb2a76812c2b89e9f4e73f454ed40be3f5a82b6b6a0f7716fd9041716f9d2f6R1169)
